bindings/rust/build.rs

```diff
@@ -86,38 +108,53 @@ fn main() {
     }
 }
 
-/// Searches for LLVM in $NYXSTONE_LLVM_PREFIX/bin and in the path and ensures proper version and static
-/// linking support.
-///
+/// Searches for LLVM in $NYXSTONE_LLVM_PREFIX/bin and in the path and ensures proper version
 /// # Returns
 /// `Ok()` and PathBuf to llvm-config, `Err()` otherwise
 fn search_llvm_config() -> Result<PathBuf> {
     let prefix = env::var(ENV_LLVM_PREFIX)
-        .map(|p| PathBuf::from(p).join("bin"))
+        .and_then(|p| {
+            if p.is_empty() {
+                return Err(std::env::VarError::NotPresent);
+            }
+
+            Ok(PathBuf::from(p).join("bin"))
+        })
         .unwrap_or_else(|_| PathBuf::new());
 
-    let llvm_config = Path::new(&prefix).join("llvm-config");
-    let version = get_major_version(&llvm_config).map_err(|_| {
-        anyhow!(
-            "No llvm-config found in {}",
-            if prefix == PathBuf::new() {
-                "$PATH"
-            } else {
-                "$NYXSTONE_LLVM_PREFIX"
-            }
-        )
-    })?;
-    let can_link = can_link_static(&llvm_config);
+    for name in llvm_config_binary_names() {
+        let llvm_config = Path::new(&prefix).join(name);
 
-    if version != "15" {
-        return Err(anyhow!("LLVM major version is {}, must be 15.", version));
-    }
+        let Ok(version) = get_major_version(&llvm_config) else {
+            continue;
+        };
 
-    if !can_link {
-        return Err(anyhow!("LLVM install does not include static libraries.",));
+        let version = version.parse::<u32>().context("Parsing LLVM version")?;
+
+        if version < 15 {
+            return Err(anyhow!("LLVM major version is {}, must be 18.", version));
```

Comment:
Check for version >= 15. Might be better to check for currently supported versions 15-18.
Currently provides the error message `must be 18`, might be better to adjust to `must be one of 15, 16, 17, 18`.